flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShim.java

```diff
@@ -33,16 +36,18 @@
 import org.apache.hadoop.hive.metastore.api.UnknownDBException;
 import org.apache.hadoop.hive.ql.udf.generic.SimpleGenericUDAFParameterInfo;
 import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
 import org.apache.thrift.TException;
 
 import java.io.IOException;
+import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
 /**
  * A shim layer to support different versions of Hive.
  */
-public interface HiveShim {
+public interface HiveShim extends Serializable {
```

Comment:
Test will fail if HiveShim is not serializable. Do you mean we should first open and merge a dedicated PR for this and come back later?

Reply:
Does the test fail before this PR, as HiveShim is not serializable before. If we need the fix to pass the test, we can put in the same PR to reduce the overhead.

Reply:
Tests pass before this PR only because we don't have enough test coverage to expose the issue. Before this PR, HiveShim is only used to get object inspectors for DATE and TIMESTAMP columns, while with the PR, HiveShim is needed for all columns. So we need the fix to pass the tests.